core/src/mock_alpenglow_consensus.rs

```diff
@@ -0,0 +1,689 @@
+        );
+    }
+
+    pub(crate) fn signal_new_slot(&mut self, slot: Slot, bank: &Bank) {
+        let Some(config) = get_test_config_from_account(bank) else {
+            debug!("Alpenglow votes disabled by on-chain config");
+            return;
+        };
+        if slot < self.highest_slot || slot < 3 {
+            trace!(
+                "Skipping AG logic for slot {slot}, current highest slot is {}",
+                self.highest_slot
+            );
+            return;
+        }
+        let interval = config.test_interval_slots as u64;
+        if interval == 0 {
+            return;
+        }
+        self.highest_slot = slot;
+        let concurrent_tests = 3u64.saturating_sub(interval) + 1;
+        assert!(1 <= concurrent_tests && concurrent_tests <= 3);
+        // since we may be running up to 3 voting processes concurrently,
+        // they will all be in their own phase
+        for ct in 0..concurrent_tests {
+            // this is currently set up to work over a course of 3 slots.
+            // it will only work correctly if we do not actually fork within those 3 slots, and
+            // are consistently casting votes for all 3.
+            let phase = (slot % interval) + ct;
+            match phase {
+                // Clear the stats for the previous slots in preparation for the new one
+                0 => {
+                    // prepare to receive votes for the next slot in advance
+                    // in case we are really late getting shreds
+                    self.prepare_to_receive(slot + 1);
+                }
+                // send Notarize votes (rest of the consensus happens based on received packets)
+                1 => {
+                    trace!("Starting voting in slot {slot}");
+                    self.command_sender.send(Command::SendNotarize(slot));
+                }
+                // report metrics
+                2 => {
+                    trace!("Reporting statistics for slot {}", slot - 1);
+                    // avoid holding locks while reporting metrics
+                    let (peers, total_staked) = {
+                        let mut lockguard = get_state_for_slot(self.state.as_slice(), slot - 1)
+                            .lock()
+                            .unwrap();
+                        let total_staked = lockguard.total_staked;
+                        let peers = lockguard.reset();
+                        (peers, total_staked)
+                    };
+                    Self::report_collected_votes(&peers, total_staked);
+                }
+                // in case we have missed previous slot, clean up and block reception
+                // but do not report stats (as they will be all garbled by now)
+                n => {
+                    for state in self.state.iter() {
+                        let mut lockguard = state.lock().unwrap();
+                        if lockguard.current_slot > 0 && (n > lockguard.current_slot + 1) {
```

Comment:
Do you want to maybe add datapoint_info! for the case where we missed previous phase? So that if tests don't generate packets as expected we know what the reason is.